packages/common/src/types/filterGrammar.ts

```diff
@@ -4,6 +4,12 @@ import { type AnyType } from './any';
 import { UnexpectedServerError } from './errors';
 import { FilterOperator, type MetricFilterRule } from './filter';
 
+export type RequiredFilter = {
+    [key: string]: AnyType;
+} & {
+    required?: boolean;
```

Comment:
since we can accept all different values within `[key: string]` , required can't be used like this:
```
{
    [key: string]: AnyType;
    required?: boolean;
}
```
There is a small edge case, which is that you can't have a required: false filter on a `required` named dimension... A workaround would be to implement an alias (like `req`) but I'd wait until this becomes an issue...